doc/rst/users-guide/datapar/promotion.rst

```diff
@@ -128,6 +128,20 @@ values:
   :language: text
   :lines: 27,25
 
+When arguments are promoted, they are only evaluated once. Consider a promoted
```

Comment:
I think the use of a multi-argument call definitely clarifies this point and that your example is a good one.

For this call you're using as an illustration, by my sense of terminology:
* the call to computeMask() is promoted (it's being called multiple times rather than once)
* the `x` and `y` arguments are being promoted (they are the arguments whose actuals are an array of the formal type)
* the 'b' argument is not being promoted even though it's part of a promoted call (its actual and formal types match)

For that reason, the wording "When arguments are promoted" doesn't seem to apply to the `computeMask()` call, which is the one you want us to focus on—it generates a `bool`, and the formal is expecting a `bool`.  Two alternate wordings that wouldn't trip me up in this way might be:

* "Any non-promoted arguments to a promoted call are only evaluated once. ...
* "The arguments to a promoted call are only evaluated once."

Where the first seems to more clearly say precisely what you're trying to highlight here, but at the downside of being wordier.  The second is simpler wording-wise, but requires mentally thinking about both cases (a promoted argument and a non-promoted argument), so may require a second sentence before transitioning into the example.

I'm also wondering whether the second sentence should be more precise about which argument is the result of the call, the fact that that argument is a non-promoted match, etc.